array_analyzer/load/report.py

```diff
@@ -1,68 +1,132 @@
-import array_analyzer.extract.constants as constants
+import collections
 from copy import deepcopy
+import logging
 import numpy as np
+import os
 import pandas as pd
 
+import array_analyzer.extract.constants as constants
 
-def write_od_to_plate(data, well_name, well_array):
-    """
-    Given data from image_parser.compute_od, write the output to an array
-        representing its WELL position
-    :param data: np.ndarray output from image_parser.compute_od
-    :param well_name: str well name
-    :param well_array: np.ndarray of 8x12 format (96 well plate)
-    :return:
-    """
-    if well_name in constants.IMAGE_TO_WELL:
-        (row, col) = constants.IMAGE_TO_WELL[well_name]
-    else:
-        raise AttributeError(f"well name {well_name} is not recognized")
-
-    well_array[row-1, col-1] = data
 
+class ReportWriter:
```

Comment:
Could be good to add a docstring for this class (here or in __init__ ?).

For example, it appears this class translates constants.antigen_array into a dataframe, but also creates dataframes for all the other properties (intensity, bg, od) -- along with some functions to assign values to those.

Reply:
You're right, this class definitely could use a docstring. One point I was grappling with was if the antigen_array translation to dataframe should happen here or in Metadata. What are your thoughs?